javase/src/main/java/net/glxn/qrgen/javase/QRCode.java

```diff
@@ -1,167 +1,269 @@
-        } catch (Exception e) {
-            throw new QRGenerationException("Failed to create QR image from text due to underlying exception", e);
-        }
-
-        return file;
-    }
-
-    @Override
-    public File file(String name) {
-        File file;
-        try {
-            file = createTempFile(name);
-            MatrixToImageWriter.writeToPath(createMatrix(text), imageType.toString(), file.toPath(), matrixToImageConfig);
-        } catch (Exception e) {
-            throw new QRGenerationException("Failed to create QR image from text due to underlying exception", e);
-        }
-
-        return file;
-    }
-
-    @Override
-    protected void writeToStream(OutputStream stream) throws IOException, WriterException {
-        MatrixToImageWriter.writeToStream(createMatrix(text), imageType.toString(), stream, matrixToImageConfig);
-    }
-
-    private File createTempSvgFile() throws IOException {
-        return createTempSvgFile("QRCode");
-    }
-
-    private File createTempSvgFile(String name) throws IOException {
-        File file = File.createTempFile(name, ".svg");
-        file.deleteOnExit();
-        return file;
-    }
-
-    public QRCode withColor(int onColor, int offColor) {
-        matrixToImageConfig = new MatrixToImageConfig(onColor, offColor);
-        return this;
-    }
+	public static final MatrixToImageConfig DEFAULT_CONFIG = new MatrixToImageConfig();
+
+	protected final String text;
+	protected MatrixToImageConfig matrixToImageConfig = DEFAULT_CONFIG;
+
+	protected QRCode(String text) {
+		this.text = text;
+		qrWriter = new QRCodeWriter();
+	}
+
+	/**
+	 * Create a QR code from the given text. <br>
+	 * <br>
+	 * <p>
+	 * There is a size limitation to how much you can put into a QR code. This has
+	 * been tested to work with up to a length of 2950 characters.<br>
+	 * <br>
+	 * </p>
+	 * <p>
+	 * The QRCode will have the following defaults: <br>
+	 * {size: 100x100}<br>
+	 * {imageType:PNG} <br>
+	 * <br>
+	 * </p>
+	 * Both size and imageType can be overridden: <br>
+	 * Image type override is done by calling {@link QRCode#to(ImageType)} e.g.
+	 * QRCode.from("hello world").to(JPG) <br>
+	 * Size override is done by calling {@link QRCode#withSize} e.g.
+	 * QRCode.from("hello world").to(JPG).withSize(125, 125) <br>
+	 *
+	 * @param text
+	 *            the text to encode to a new QRCode, this may fail if the text is
+	 *            too large. <br>
+	 * @return the QRCode object <br>
+	 */
+	public static QRCode from(String text) {
+		return new QRCode(text);
+	}
+
+	/**
+	 * Creates a a QR Code from the given {@link VCard}.
+	 * <p>
+	 * The QRCode will have the following defaults: <br>
+	 * {size: 100x100}<br>
+	 * {imageType:PNG} <br>
+	 * <br>
+	 * </p>
+	 * 
+	 * @param vcard
+	 *            the vcard to encode as QRCode
+	 * @return the QRCode object
+	 */
+	public static QRCode from(VCard vcard) {
+		return new QRCode(vcard.toString());
+	}
+
+	public static QRCode from(BizCard bizCard) {
```

Comment:
Instead of adding a new `from` method for every scheme, we could have a `from(Scheme)` instead. 
You could introduce a Scheme interface that VCard and all other schemes implement. The Scheme interface would need to expose amethod to get the string value.